web/src/message_edit_history.ts

```diff
@@ -18,30 +21,61 @@ import * as timerender from "./timerender";
 import * as ui_report from "./ui_report";
 import {user_settings} from "./user_settings";
 
-export function fetch_and_render_message_history(message) {
-    channel.get({
+type ContentEditHistoryItem = {
+    timestamp: string;
+    display_date: string;
+    show_date_row: boolean;
+    edited_by_notice: string;
+    body_to_render?: string;
+    topic_edited?: boolean;
+    prev_topic?: string;
+    new_topic?: string;
+    stream_changed?: boolean;
+    prev_stream?: string;
+};
+
+const fetch_and_render_message_history_response_schema = z.object({
+    message_history: z.array(
+        z.object({
+            content: z.string(),
+            rendered_content: z.string(),
+            timestamp: z.number(),
+            topic: z.string(),
+            user_id: z.number().or(z.null()),
+            prev_topic: z.string().optional(),
+            stream: z.number().optional(),
+            prev_stream: z.number().optional(),
+            prev_content: z.string().optional(),
+            prev_rendered_content: z.string().optional(),
+            content_html_diff: z.string().optional(),
+        }),
+    ),
+});
+
+export function fetch_and_render_message_history(message: Message): void {
+    void channel.get({
         url: "/json/messages/" + message.id + "/history",
         data: {message_id: JSON.stringify(message.id)},
         success(data) {
-            const content_edit_history = [];
+            const clean_data = fetch_and_render_message_history_response_schema.parse(data);
```

Comment:
I think this could be called `server_message_history_schema.parse`; not sure why we need such a long name. The `fetch` part in `fetch_and_render_message_history` is there because this function does an API request to fetch data from the server...